_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -0,0 +1,140 @@
+due to binary incompatibility. This inability to upgrade any packages without breaking anything is common known as **Dependency Hell**.
+
+![Dependency Hell]({{ site.baseurl }}/resources/images/library-author-guide/dependency_hell.png)
+
+How can we, as library authors, spare our users of runtime errors and dependency hell?
+
+* Use **Migration Manager** (MiMa) to catch unintended binary compatibility breakages before releasing a new library version
+* **Avoid breaking binary compatibility** through careful design and evolution of your library interfaces
+* Communicate binary compatibility breakages clearly through **versioning**
+
+## MiMa - Check Binary Compatibility with Previous Library Versions 
+
+The [Migration Manager for Scala](https://github.com/typesafehub/migration-manager) (MiMa) is a tool for diagnosing binary incompatibilities between different library versions.
+
+When run standalone in the command line, it will compare the .class files in the two provided JARs and report any binary incompatibilities found. Most library authors use the [SBT plugin](https://github.com/typesafehub/migration-manager/wiki/Sbt-plugin)
+to help spot binary incompatibility between library releases. (Follow the link for instructions on how to use it in your project)
+
+## Designing for Evolution - without breaking binary compatibility
+
+TODO
+
+## Versioning Scheme - Communicate binary compatiblity breakages
+
+We recommend using the following schemes to communicate binary and source compatibility to your users:
+
+* Any release with the same major version are **Binary Backwards Compatible** with each other
+* A minor version bump signals new features and **may contain minor source incompatibilities** that can be easily fixed by the end user
+* Patch version for bugfixes and minor behavioural changes
+* For **expreimental library versions** (where the major version is `0`, such as `v0.1.0`), a minor version bump **may contain both source and binary breakages**
+* Some libraries may take a harder stance on maintaining source compatibility, bumping the major version number for ANY source incompatibility even if they are binary compatible
+
+Some examples:
+
+* `v1.0.0 -> v2.0.0` is <span style="color: red">binary incompatible</span>. Cares needs to be taken to make sure no evicted versions are still in the `v1.x.x` range to avoid runtime errors
+* `v1.0.0 -> v1.1.0` is <span style="color: blue">binary compatible</span> and maybe source incompatible
+* `v1.0.0 -> v1.0.1` is <span style="color: blue">binary compatible</span> and source compatible
+* `v0.4.0 -> v0.5.0` is <span style="color: red">binary incompatible</span> and maybe source incompatible
+* `v0.4.0 -> v0.4.1` is <span style="color: blue">binary compatible</span> and source compatible
+
+Many libraries in the Scala ecosystem has adopted this versioning scheme. A few examples are [Akka](http://doc.akka.io/docs/akka/2.5/scala/common/binary-compatibility-rules.html),
```

Comment:
re:Akka: Not really, the above explained scheme is much much weaker than what Akka guarantees.

Here it said:
`+* `v0.4.0 -> v0.5.0` is <span style="color: red">binary incompatible</span> and maybe source incompatible`
In Akka this would be fully binary compatible, we do not talk about source compatibility, but usually it would be kept as well. Only deprecated things (that were such for at least 2 full release cycle) could be removed in such release

Reply:
Ah, sorry! I see that you mean the `0.` prefix here.